tiled/server/app.py

```diff
@@ -113,10 +112,11 @@ def build_app(
     tree,
     authentication=None,
     server_settings=None,
-    query_registry=None,
-    serialization_registry=None,
-    compression_registry=None,
-    validation_registry=None,
+    query_registry: Optional[QueryRegistry] = None,
```

Comment:
I wonder if there's any reason not to make the default values the default registry objects, and drop the `Optional`.

Reply:
Is `build_app` ever called not with the result of `construct_build_app_kwargs`? Just thinking about where the defaults should be injected.

Although ideally I'd like to get rid of `construct_build_app_kwargs` and anything that uses kwargs...

Reply:
Yes, a whole bunch in the test suite. One example at random:

https://github.com/bluesky/tiled/blob/ed9033801890f3db735e76915bf6dec96726890a/tiled/_tests/test_distinct.py#L65

Reply:
I'm going to push back on changing it for now, because I'd like to rip out construct_build_app_kwargs in the longer run, make build_app take just tree and settings, and eventually settings gets created on startup exactly once.